vscode-dotnet-runtime-library/src/test/unit/DotnetCoreAcquisitionWorker.test.ts

```diff
@@ -213,4 +213,11 @@ suite('DotnetCoreAcquisitionWorker Unit Tests', function() {
         const events = eventStream.events.filter(event => event instanceof DotnetAcquisitionStarted);
         assert.equal(events.length, 1);
     });
+
+    test('Get Expected Path', async () => {
+        const [acquisitionWorker, eventStream, context] = getTestAcquisitionWorker(true);

```

Comment:
I like the use of the `testAcquisitionWorker`, I think that will indeed use the actual install script writer to write the install script out. But I think this will not test the issue the user was facing by having ' in the path.

You could make a mock IInstallationDirectoryProvider that writes to a path with ' such as the users path in the issue. But the install script acquisition worker does not write to the directory provided by the installationdirectoryprovider, so you'd have to change that as well. 

Instead, I would recommend looking into `InstallScriptAcquisitionWorker` and making a mock of it that lets you change the `scriptFilePath,` and then change that to include the apostrophe in the install path to test it ;)